invokeai/backend/stable_diffusion/diffusion/unet_attention_patcher.py

```diff
@@ -25,10 +33,21 @@ def _prepare_attention_processors(self, unet: UNet2DConditionModel):
                 # "attn1" processors do not use IP-Adapters.
                 attn_procs[name] = CustomAttnProcessor2_0()
             else:
+                ip_adapter_attention_weights = IPAdapterAttentionWeights(ip_adapter_weights=[], skip=False)
+                for ip_adapter in self._ip_adapters:
+                    ip_adapter_weight = ip_adapter["ip_adapter"].attn_weights.get_attention_processor_weights(idx)
+                    skip = True
+                    for block in ip_adapter["target_blocks"]:
+                        if block in name:
+                            skip = False
+                            break
+                    ip_adapter_attention_weights.ip_adapter_weights = [ip_adapter_weight]
+                    ip_adapter_attention_weights.skip = skip
```

Comment:
We should be appending to these values, not overwriting them in each iteration. (See my comment on the `IPAdapterAttentionWeights` type).

Reply:
Oh man .. That explains why I wasn't getting the correct array length. So the padding wasn't actually needed. Okay. I fixed that up. And now the results are actually even better.